src/lib/rnp.cpp

```diff
@@ -6421,6 +6421,28 @@ FFI_GUARD
 
 rnp_result_t
 rnp_key_valid_till(rnp_key_handle_t handle, uint32_t *result)
+try {
+    if (!result) {
+        return RNP_ERROR_NULL_POINTER;
+    }
+    uint64_t     res = 0;
+    rnp_result_t ret = rnp_key_valid_till64(handle, &res);
+    if (ret) {
+        return ret;
+    }
+    if (res == UINT64_MAX) {
+        *result = UINT32_MAX;
+    } else if (res > UINT32_MAX) {
+        *result = UINT32_MAX - 1;
```

Comment:
Why the `- 1`?

Reply:
Because UINT32_MAX is reserved for 'never expires'.

Reply:
Then you should probably do: `if (res >= UINT32_MAX)` to catch the case that `res` happens to equal `UINT32_MAX`.

Reply:
Agree, missed that. Thanks.